dask/tests/test_order.py

```diff
@@ -351,7 +351,7 @@ def test_local_parents_of_reduction(abcde):
     b1, b2, b3 = [b + i for i in "123"]
     c1, c2, c3 = [c + i for i in "123"]
 
-    expected = [a3, a2, a1, b3, b2, b1, c3, c2, c1]
+    expected = [a3, b3, c3, a2, a1, b2, b1, c2, c1]
```

Comment:
Before the synchronous scheduler did not submit all ready work immediately, but only submitted as many tasks as it had workers available. This was a side-effect of limiting the number of tasks submitted to the number of workers available. As the synchronous scheduler only ever had 1 worker, it would just submit `a3` and reassess. Hence why it went to `a2`. However this was not the case for the threaded or multiprocessing cases as they had more workers and could submit more of the ready tasks.

Now in all cases (including the synchronous scheduler), we submit all work that is ready to go as soon as it is ready. This is because `Executor`s have their own queues and can manage how much work they schedule vs. keep queued. We still submit tasks in the order requested, but if it is ready to go we still submit it. So `a3`, `b3`, and `c3` are all ready to go on the first pass. Therefore we submit all of them. We still prioritize doing `a2` and `a1` next. As result `b2` still comes after `a1` as the test intended. This is the same behavior we would have seen for threading and multiprocessing both before this change and after it.

In any event hopefully this gives more context. With this change the synchronous scheduler should be more consistent when consider in relation to the other schedulers.